src/main/java/uk/gov/register/indexer/IndexDriver.java

```diff
@@ -1,48 +1,59 @@
 package uk.gov.register.indexer;
 
-import uk.gov.register.core.*;
+import uk.gov.register.core.Entry;
+import uk.gov.register.core.Item;
+import uk.gov.register.core.Record;
 import uk.gov.register.indexer.function.IndexFunction;
 import uk.gov.register.store.DataAccessLayer;
-import uk.gov.register.configuration.IndexFunctionConfiguration.IndexNames;
 
 import java.util.*;
 import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
 
 public class IndexDriver {
-    private final DataAccessLayer dataAccessLayer;
+    private final Map<String, AtomicInteger> currentIndexEntryNumbers;
 
-    public IndexDriver(DataAccessLayer dataAccessLayer) {
-        this.dataAccessLayer = dataAccessLayer;
+    public IndexDriver() {
+        this.currentIndexEntryNumbers = new HashMap<>();
     }
 
-    public void indexEntry(RegisterReadOnly register, Entry entry, IndexFunction indexFunction) {
-        Optional<Record> currentRecord = (entry.getEntryType() == EntryType.user)
-                ? register.getRecord(entry.getKey())
-                : register.getDerivationRecord(entry.getKey(), IndexNames.METADATA);
+    public void indexEntry(DataAccessLayer dataAccessLayer, Entry entry, IndexFunction indexFunction,
```

Comment:
It might be better to keep `DataAccessLayer` as part of the constructor rather than as an argument to this method?